File: app/src/main/java/com/codemybrainsout/imageviewer/utility/PrefHelper.java
package com.codemybrainsout.imageviewer.utility;

import android.content.Context;
import android.content.SharedPreferences;

import com.google.gson.Gson;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class PrefHelper {


    public PrefHelper() {
        super();
    }

    public static final String PREF = "Resplash";


    public static void storeList(Context context, String key, List list) {

        SharedPreferences settings;
        SharedPreferences.Editor editor;
        settings = context.getSharedPreferences(PREF, Context.MODE_PRIVATE);
        editor = settings.edit();
        Gson gson = new Gson();
        String jsonFavorites = gson.toJson(list);
        editor.putString(key, jsonFavorites);
        editor.commit();
    }

    public static ArrayList loadList(Context context, String key) {

        SharedPreferences settings;
        List favorites;
        settings = context.getSharedPreferences(PREF, Context.MODE_PRIVATE);
        if (settings.contains(key)) {
            String jsonFavorites = settings.getString(key, null);
            Gson gson = new Gson();
            String[] favoriteItems = gson.fromJson(jsonFavorites, String[].class);
            favorites = Arrays.asList(favoriteItems);
            favorites = new ArrayList(favorites);
        } else
            return new ArrayList();
        return (ArrayList) favorites;
    }


    public static void deleteList(Context context) {

        SharedPreferences myPrefs = context.getSharedPreferences(PREF,
                Context.MODE_PRIVATE);
        SharedPreferences.Editor editor = myPrefs.edit();
        editor.clear();
        editor.commit();
    }
}

File: app/src/main/java/com/codemybrainsout/imageviewer/utility/PrefHelper.kt
package com.codemybrainsout.imageviewer.utility

import android.content.Context
import android.content.SharedPreferences
import com.google.gson.Gson
import java.util.*

object PrefHelper {

    const val PREF = "Resplash"

    @JvmStatic
    fun storeList(context: Context, key: String, list: List<String>) {
        val editor: SharedPreferences.Editor
        val settings: SharedPreferences = context.getSharedPreferences(PREF, Context.MODE_PRIVATE)
        editor = settings.edit()
        val gson = Gson()
        val jsonFavorites = gson.toJson(list)
        editor.putString(key, jsonFavorites)
        editor.apply()
    }

    @JvmStatic
    fun loadList(context: Context, key: String): ArrayList<String> {
        var favorites: List<String>?
        val settings: SharedPreferences = context.getSharedPreferences(PREF, Context.MODE_PRIVATE)
        if (settings.contains(key)) {
            val jsonFavorites = settings.getString(key, null)
            val gson = Gson()
            val favoriteItems = gson.fromJson(jsonFavorites, Array<String>::class.java)
            favorites = listOf(*favoriteItems)
            favorites = ArrayList(favorites)
        } else return ArrayList<String>()
        return favorites
    }

    fun deleteList(context: Context) {
        val myPrefs = context.getSharedPreferences(
            PREF,
            Context.MODE_PRIVATE
        )
        val editor = myPrefs.edit()
        editor.clear()
        editor.apply()
    }
}